tests/rip7560/test_validation_rules.py

```diff
@@ -98,7 +111,7 @@ def entity_to_contract_name(entity):
     return None
 
 
-def get_build_func(entity, rule):
+def get_build_func(entity, rule, assert_func, stake_manager):
```

Comment:
it was ugly, and it only gets worse
there is now a lot of code in creating the tests. it was supposed to be parameterised table with no logic